core/schema/sdk.go

```diff
@@ -469,19 +469,15 @@ func (sdk *goSDK) baseWithCodegen(
 	}
 
 	if err := sdk.dag.Select(ctx, ctr, &ctr, dagql.Selector{
-		Field: "withNewFile",
+		Field: "withMountedFile",
```

Comment:
Just curious: is there any reason not to use `withFile` here? I recall discussions around possibly getting rid of `withMountedXXX` since the normal versions already do the efficient MergeOp technique under the hood.

Reply:
MergeOp can only be used directly when copying from `/` to `/`, otherwise there needs to be an intermediate copy step to align everything to be at `/`. There's ways of fixing that upstream (basically, use the same hardlink approach of mergeop but for all copy-ops too), but until then `WithMountedX` is still gonna be more efficient in the general case (though this only really becomes noticeable once you get to like thousands of files and/or files greater than 100MB).